### PR TITLE
Delete importer pod on pvc delete request

### LINKED ISSUE #525: Referenced POD object not deleted if failing / in retry-loop
/kind bug

**What happened**:
When you delete a DataVolume and a POD is in a retry/crash loop, the PVC goes to Terminating and is stuck there forever as the POD is never deleted and just continues retrying forever.

You CAN address this by issuing a kubectl delete against the POD, on the next restart interval the update will be detected and the POD finally destroyed, leading to deletion of the PVC completing as well.

**What you expected to happen**:
Normally referenced objects are deleted as part of the deletion of their parent, regardless of the retry loops on the POD I would expect it to be deleted, then the PVC deleted and then finally the DataVolume deleted. 

**How to reproduce it (as minimally and precisely as possible)**:

Create a DataVolume with an invalid URL, thereby causing the importer POD to go into it's restart-on-failure loop.

Now, issue a delete of the Data Volume ``kubectl delete dv my-data-volume``

You'll note that the DataVolume is removed, the PVC goes to terminating but the POD just keeps on doing it's restarts.

**Anything else we need to know?**:
Related issues:
#523 

Also opening another issue to rework our retry strategy 

**Environment**:
- CDI version Current Master

### LINKED ISSUE #649: Can't delete PVC when importer pods are running/restarting
/kind bug

**What happened**:
Related to issue #525 we have a similar situation when not using DataVolumes.  The user requests a PVC using the importer, if the Importer POD is in a retry loop and a user requests deletion of the PVC the PVC will be stuck in terminating until the user forcibly deletes the PODs that are retrying forever.

**What you expected to happen**:
This is what I expect, but it is not ideal from a usability perspective.  Instead, we should go ahead and delete the retrying PODs for the user when we detect the PVC delete request.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a PVC that uses the importer, but provide an invalid URL 
2. Wait until the POD enters the retry/restart loop
3. Issue a PVC delete no the importer created PVC

### PR DESCRIPTION
**Pull request by j-griffith**
Related to issue #525, we have a similar situation when using PVCs
without DataVolumes.  The user requests a PVC with CDI Importer
annotations, if the Importer POD is in a fail/retry loop and the user
requests deletion of the PVC the PVC will be stuck in terminating
waiting for the user to forcibly delete the running Import POD.

This PR adds a check for PVC delete in the import controller and
will delete the POD explicitly if it sees a PVC delete request.  This
will also be picked up for Import DataVolumes, so we can remove the
extra checks that were previously added in the DV controller.

fixes #649
fixes #525

```release-note
NONE
```

### CODE REVIEW

**Comment by j-griffith** [comment]
> So far so good but I have a couple questions:
> 
> The controller is not watching for PVC delete events so I think it may be possible to miss deletes. Fine for kubectl because it does a foreground delete which sets the deletion timestamp and adds a finalizer 

True, I was focused on the reported issue of the user not "liking" the default behavior of kubectl, wondering what non-kubectl use case we'd have here?  Anything done by our controllers can set the delete policy to do what we want explicitly to avoid any issues.

> but the object is not really deleted until all the children are. So I wonder if we have an issue with background deletes (which are the default)? If so, cloe/uplod controllers are also affected.

Yes, this focuses on just import for now, but we'd need the same changes to be added to clone and upload as well if we agree on an approach here.

> I believe the code in the datavolume controller that deletes pods should be removed along with this.

Agreed, I'll look at that; I think it needs to be modified; not completely removed. I believe we'll need to do something like delete the PVC so then the other controllers with this addition will clean everything up properly.

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -81,8 +83,6 @@ func NewController(client kubernetes.Interface,
 			}
 			c.handlePodUpdate(new)
 		},
-
-		DeleteFunc: c.handlePodDelete,
```
Comment (mhenriks) [inline pkg/controller/controller.go]: Why get rid of this?  If some jerk deletes a pod, shouldn't the controller recreate it?

Comment (j-griffith) [reply pkg/controller/controller.go]: Oops!  That was accidental, nice catch!

**Comment by j-griffith** [comment]
Rebase is KILLING ME

**Comment by awels** [comment]
> Rebase is KILLING ME

Welcome to software engineering?

**Comment by j-griffith** [comment]
Looks like something upstream in coveralls went wonky:  
https://github.com/lemurheavy/coveralls-public/issues/1264 (and a number of duplicates)

**Comment by j-griffith** [comment]
Seeing these same failures on master currently as well:  https://hastebin.com/mepulajima.cpp Looking

**Comment by j-griffith** [comment]
ci test please

**Pull request merged**